W-11862525-first-draft-for-beta-site-build-CP (#114)
mulesoft/docs-apikit: modules/ROOT/pages/apikit-for-soap-module-reference.adoc

= APIkit for SOAP Module Reference
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]

== Configurations

=== Config

Configuration element for a *SOAP Router*

==== Parameters

[cols="1a,2a,3a,4a,5a"]
|===
| Name | Type | Description | Default Value | Required
| Name | String | Connectors use this configuration name to reference its configuration. | | *x*
| Soap Version | Enumeration, one of:

* SOAP11
 
* SOAP12 

| SOAP version of the WSDL. |  | 
| Address | String | Address of the web service. |  | 
| MTOM Enabled | Boolean | Defines if the MTOM protocol manages the attachments. | `false` | 

> | Inbound Validation Enabled | Boolean | Defines if the MTOM protocol manages the attachments. | `false` | 

Reviewer: The description of this item is the same as the previous one, although the parameters are different. Please check if this is an error.

Author: OK! Thanks! 
Requesting assistance from SMEs.

Author: Great catch! It seems the table got broken somehow. 
Thanks!

| HTTP Status Var Name | String | Variable name for the HTTP status. | `httpStatus` | 

> | Namespace Prefixes | Object |  |  | 

Reviewer: No description for this parameter?

Author: OK! Thanks! 
Requesting assistance from SMEs.

Author: Description added. Thanks!

> | Expiration Policy | xref:apikit-for-soap-module-reference.adoc#expiration-policy[Expiration Policy] | Configures the minimum amount of time that a dynamic configuration instance can remain idle before the runtime considers it eligible for expiration. The platform does not necessarily expire the instance at the exact moment that it becomes eligible. The runtime actually purges the instances when it sees it fit. |  | 

Reviewer: ```suggestion
| Expiration Policy | xref:apikit-for-soap-module-reference.adoc#expiration-policy[Expiration Policy] | Configures the minimum amount of time that a dynamic configuration instance can remain idle before the runtime considers it eligible for expiration. The platform does not necessarily expire the instance at the exact moment that it becomes eligible. The runtime purges the instances when it considers it appropriate. |  | 
```

Author: OK! Thanks!

| WSDL Location | String | WSDL file URL, remote or local. |  | *x*
| Service | String | Name of the service. |  | *x*
| Port | String | Name of the port. |  | *x*
|===

==== Associated Operations

* xref:apikit-for-soap-module-reference.adoc#soap-fault[Fault]

* xref:apikit-for-soap-module-reference.adoc#soap-router[Router]

== Operations

=== SOAP Fault

`<apikit-soap:fault>`

APIkit for SOAP Fault

==== Parameters

[cols="1a,2a,3a,4a,5a", options="header"]
|===
| Name | Type | Description | Default Value | Required
| Configuration | String | Name of the configuration. | | *x*
| Payload | Binary |  |  #[payload] | 
| Streaming Strategy | 

* xref:apikit-for-soap-module-reference.adoc#repeatable-in-memory-stream[Repeatable in memory stream]

* xref:apikit-for-soap-module-reference.adoc#repeatable-file-store-stream[Repeatable file store stream]

* Non repeatable stream | Configure if repeatable streams and their behavior are used. |  | 
| Operation | String |  |  | *x*

> | Fault | String |  |  | *x*

Reviewer: No descriptions?

Author: OK! Thanks! 
Requesting assistance from SMEs.

Author: Descriptions added. Thanks!

| Target Variable | String | Name of a variable on which the operation output is placed. |  | 

> | Target Value | String | An expression that is evaluated against the operation output and the outcome of that expression is stored in the target variable. |  #[payload] | 

Reviewer: ```suggestion
| Target Value | String | Expression that is evaluated against the operation output. The outcome of the expression is stored in the target variable. |  #[payload] | 
```

Author: OK! Thanks!

|===

==== Output

|===
| *Type* | Binary
|===

==== For Configurations

* xref:apikit-for-soap-module-reference.adoc#config[Config]

==== Throws

* APIKIT-SOAP:FAULT 
* APIKIT-SOAP:ROUTER 

=== SOAP Router

`<apikit-soap:router>`

APIkit for SOAP Router

==== Parameters

[cols="1a,2a,3a,4a,5a", options="header"]
|===
| Name | Type | Description | Default Value | Required
| Configuration | String | Name of the configuration. | | *x*
| Message | Binary |  |  #[payload] | 

> | Attributes | Object |  |  | *x*

Reviewer: No description?

Author: OK! Thanks! 
Requesting assistance from SMEs.

Author: Descriptions added. Thanks!

| Output Mime Type | String | MIME type of the payload that this operation outputs. |  | 
| Output Encoding | String | Encoding of the payload that this operation outputs. |  | 
| Streaming Strategy | 

* xref:apikit-for-soap-module-reference.adoc#repeatable-in-memory-stream[Repeatable in memory stream]

* xref:apikit-for-soap-module-reference.adoc#repeatable-file-store-stream[Repeatable file store stream]

* Non repeatable stream | Configure if repeatable streams and their behavior are used. |  | 
| Target Variable | String | Name of a variable on which the operation output is placed. |  | 

> | Target Value | String | Expression that is evaluated against the operation output and the outcome of that expression is stored in the target variable. | #[payload] | 

Reviewer: ```suggestion
| Target Value | String | Expression that is evaluated against the operation output. The outcome of the expression is stored in the target variable. |  #[payload] | 
```

Author: OK! Thanks!

| *Attributes Type* | xref:apikit-for-soap-module-reference.adoc#soap-attributes[SOAP Attributes]
|===

==== For Configurations

* xref:apikit-for-soap-module-reference.adoc#config[Config]

==== Throws

* APIKIT-SOAP:FAULT 
* APIKIT-SOAP:ROUTER 

== Types

=== Expiration Policy

[cols="1a,2a,3a,4a,5a", options="header"]
|===
| Field | Type | Description | Default Value | Required
| Max Idle Time | Number | A scalar time value for the maximum amount of time a dynamic configuration instance is allowed to be idle before it's considered eligible for expiration |  | 
| Time Unit | Enumeration, one of:

* NANOSECONDS

* MICROSECONDS

* MILLISECONDS

* SECONDS

* MINUTES

* HOURS

* DAYS | A time unit that qualifies the `maxIdleTime` attribute |  | 
|===

=== Repeatable in Memory Stream

[cols="1a,2a,3a,4a,5a", options="header"]
|===
| Field | Type | Description | Default Value | Required
| Initial Buffer Size | Number | Amount of memory that is allocated to consume the stream and provide random access to it. If the stream contains more data than buffer capacity, then it is expanded according to the `bufferSizeIncrement` attribute, with an upper limit of `maxInMemorySize`. |  | 

> | Buffer Size Increment | Number | How much buffer size expands if it exceeds its initial size. A value of zero or lower specifies that the buffer does not expand. When the buffer is full, `STREAM_MAXIMUM_SIZE_EXCEEDED` error occurs. |  | 

Reviewer: ```suggestion
| Buffer Size Increment | Number | Amount the buffer size expands if it exceeds its initial size. A value of zero or lower specifies that the buffer does not expand. When the buffer is full, a `STREAM_MAXIMUM_SIZE_EXCEEDED` error occurs. |  | 
```

Author: OK! Thanks!

> | Max Buffer Size | Number | The maximum amount of memory used. If more is used, a `STREAM_MAXIMUM_SIZE_EXCEEDED` error is raised. A value lower or equal to zero means no limit. |  | 

Reviewer: ```suggestion
| Max Buffer Size | Number | Maximum amount of memory used. If more is used, a `STREAM_MAXIMUM_SIZE_EXCEEDED` error is raised. A value lower or equal to zero means no limit. |  | 
```

Author: OK! Thanks!
